pkg/cmd/release-payload-controller/batchjob_controller.go

```diff
@@ -0,0 +1,279 @@
+	ReleaseCreationJobPendingMessage = "Release creation job pending"
+
+	// ReleaseCreationJobFailureMessage release creation job failure message
+	ReleaseCreationJobFailureMessage = "Release creation Job failed"
+
+	// ReleaseCreationJobSuccessMessage release creation job success message
+	ReleaseCreationJobSuccessMessage = "Release creation Job completed"
+)
+
+// BatchJobStatusController is responsible for watching batchv1.Jobs, in the job-namespace, and
+// updating the respective ReleasePayload with the status, of the job, when it completes.
+// The BatchJobStatusController watches for changes to the following resources:
+//   - batchv1.Jobs
+// The reads the following pieces of information:
+//   - .status.releaseCreationJobResult.coordinates.name
+//   - .status.releaseCreationJobResult.coordinates.namespace
+// and write the following information:
+//   - .status.releaseCreationJobResult.status
+type BatchJobStatusController struct {
+	releasePayloadNamespace string
+	releasePayloadLister    releasepayloadlister.ReleasePayloadLister
+	releasePayloadClient    releasepayloadclient.ReleaseV1alpha1Interface
+
+	batchJobNamespace string
+	batchJobLister    batchv1listers.JobLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBatchJobStatusController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	batchJobNamespace string,
+	batchJobInformer batchv1informers.JobInformer,
+	eventRecorder events.Recorder,
+) (*BatchJobStatusController, error) {
+	c := &BatchJobStatusController{
+		releasePayloadNamespace: releasePayloadNamespace,
+		releasePayloadLister:    releasePayloadInformer.Lister(),
+		releasePayloadClient:    releasePayloadClient,
+		batchJobNamespace:       batchJobNamespace,
+		batchJobLister:          batchJobInformer.Lister(),
+		eventRecorder:           eventRecorder.WithComponentSuffix("batchjob-status-controller"),
+		queue:                   workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BatchJobStatusController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, releasePayloadInformer.Informer().HasSynced)
+	c.cachesToSync = append(c.cachesToSync, batchJobInformer.Informer().HasSynced)
+
+	batchJobFilter := func(obj interface{}) bool {
+		if batchJob, ok := obj.(*batchv1.Job); ok {
+			if _, ok := batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+				return true
+			}
+		}
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			if batchJob, ok := tombstone.Obj.(*batchv1.Job); ok {
+				if _, ok = batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	batchJobInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
```

Comment:
missing informer handler for the releasepayload in case someone naughty deletes the status.

Reply:
Done.